### PR TITLE
Document custom Elixir instrumentation

### PR DESCRIPTION
**Pull request by tombruijn**
Add documentation for function decorators and instrumentation helpers.

### CODE REVIEW

**Comment by jeffkreeftmeijer** [comment]
This looks really great, so far! :+1:

**Comment by tombruijn** [comment]
Great! I just have to test some assumptions I did on the Phoenix channels bit.

Maybe add some screenshots like I did for the Ruby page: http://docs.appsignal.com/ruby/instrumentation/instrumentation.html
Note to self: this is automated in the screenshots rake task

**Comment by tombruijn** [comment]
PR updated with instrumentation helpers section. Based it largely on the decorators section, just some small differences between the scenarios. Downside is that if one section is updated, the person updated it should mind the other section as well.

@jeffkreeftmeijer can you review again?

File: lib/appsignal_markdown.rb [hunk 1]
```diff
@@ -26,8 +26,20 @@ def paragraph(text)
 
   # Add anchor tags to every heading.
   # Create a link from the heading.
+  #
+  # Extra logic added:
```
Comment (wesoudshoorn) [inline lib/appsignal_markdown.rb]: I am assuming this behavior was already there, but this is just a nice helper?

File: source/elixir/instrumentation/instrumentation.html.md [hunk 2]
```diff
@@ -2,32 +2,386 @@
 title: "Custom instrumentation for Elixir"
 ---
 
-## Elixir instrumentation
+In order to find out what specific pieces of code are causing performance
+problems it's useful to add custom instrumentation to your application. This
+allows us to create better breakdowns of which code runs slowest and what type
+of action was the most time spent on.
```
Comment (wesoudshoorn) [inline source/elixir/instrumentation/instrumentation.html.md]: `was the most time spent on` doesn't flow nicely

File: source/elixir/instrumentation/instrumentation.html.md [hunk 3]
```diff
@@ -2,32 +2,386 @@
...
 
-Using the `Appsignal.Instrumentation.Decorators` Decorators it's possible to
-add custom instrumentation to your Elixir applications.
+Custom instrumentation is possible in two ways: with function decorators and
```
Comment (wesoudshoorn) [inline source/elixir/instrumentation/instrumentation.html.md]: maybe replace `with function ...` with `through function ...`?

File: source/elixir/instrumentation/instrumentation.html.md [hunk 4]
```diff
@@ -2,32 +2,386 @@
...
+  - [Transactions](#helper-transactions)
+  - [Transaction metadata](#helper-transaction-metadata)
+  - [Namespaces](#helper-namespaces)
+      - [Custom namespaces](#helper-custom-namespaces)
+  - [Exception handling](#helper-exception-handling)
+  - [Example](#helper-example)
+
+## Function decorators
+
+Using the `Appsignal.Instrumentation.Decorators` decorator module it's possible
```
Comment (wesoudshoorn) [inline source/elixir/instrumentation/instrumentation.html.md]: Are we supposed to use a `,` after `module`?

File: source/elixir/instrumentation/instrumentation.html.md [hunk 5]
```diff
@@ -2,32 +2,386 @@
...
+  end
+end
+```
+
+###^decorator Namespaces
+
+In order to differentiate between HTTP requests and background jobs we can pass
+a namespace to the transaction once we start it.
+
+The following two namespaces official namespaces supported by AppSignal.
```
Comment (wesoudshoorn) [inline source/elixir/instrumentation/instrumentation.html.md]: Incorrect sentence structure.

File: lib/appsignal_markdown.rb (hunk 1 above)
Comment (tombruijn) [inline lib/appsignal_markdown.rb]: The first item in this list of extra logic was already added, but the other logic with the caret symbol (^) wasn't yet. That was added in this PR.

**Comment by tombruijn** [comment]
@wesoudshoorn feekback fixed! Thanks :)

**Pull request merged**